src/frontends/tensorflow_common/src/op/binary_op.cpp

```diff
@@ -57,8 +58,16 @@ OutputVector translate_floor_div_op(const NodeContext& node) {
         if (out_type.is_integral()) {
             auto float_x = make_shared<v0::Convert>(x, element::f32);
             auto float_y = make_shared<v0::Convert>(y, element::f32);
-            return make_shared<v0::Convert>(make_shared<v0::Floor>(make_shared<v1::Divide>(float_x, float_y)),
-                                            out_type);
+            auto divide_node = make_shared<v1::Divide>(float_x, float_y);
+            auto floor_node = make_shared<v0::Floor>(divide_node);
+            auto convert_node = make_shared<v0::Convert>(floor_node, out_type);
+            disable_fp16_compression(float_x);
+            disable_fp16_compression(float_y);
+            disable_fp16_compression(divide_node);
+            disable_fp16_compression(floor_node);
+            disable_fp16_compression(convert_node);
```

Comment:
these are strange changes a bit. During FE conversion we work with the original type. Here I don't think we need any compression specific logic

Reply:
I agree that it looks a bit strange. Reason i tried such solution is that we already had a similar thing for random normal
https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/common/src/random_normal_helper.cpp#L60-L62
But still it's strange, and moreover for general solution we should keep precision for all FWs.

I am currently rewriting this solution, in the latest version i call a separate pass inside `save_model`. I will write when it's ready

Reply:
@rkazants rewritten solution. Please review